game.py

```diff
@@ -215,13 +231,81 @@ def _convert_valid_actions(
         type used in pyhanabi.
         """
-        # TODO: implement this method
+        actions = []
+        for action in legal_actions:
+            if action[0] == 1:  # color hint
+                move_type = Action.ActionType.HINT_COLOR
+            elif action[0] == 2:  # rank hint
+                move_type = Action.ActionType.HINT_NUMBER
+            elif action[0] == 3:  # play
+                move_type = Action.ActionType.PLAY
+            elif action[0] == 4:  # discard
+                move_type = Action.ActionType.DISCARD
+            else:
+                continue
+
+            if move_type in {
+                Action.ActionType.HINT_COLOR,
+                Action.ActionType.HINT_NUMBER,
+            }:
+                for card in action[4]:
+                    actions.append(
+                        Action(
+                            action_type=move_type,
+                            pnr=action[1],
+                            col=COLOR_INT_CONVERSION_DICT[action[5]],
+                            num=action[6],
+                            cnr=card,  # TODO: check this
+                        )
+                    )
+            else:
+                actions.append(
+                    Action(
+                        action_type=move_type,
+                        pnr=action[1],
+                        col=COLOR_INT_CONVERSION_DICT[action[5]],
+                        num=action[6],
+                        cnr=None,  # TODO: check this
+                    )
+                )
+        return actions
 
     def _convert_action(self, native_action: Action) -> HANASIM_ACTION:
         """
         Convert a pyhanabi Action object to HanaSim's action representation.
         """
-        # TODO: implement this method
+        a_type = native_action.action_type
+        if a_type == Action.ActionType.HINT_COLOR:
+            move_type = 1
+        elif a_type == Action.ActionType.HINT_NUMBER:
+            move_type = 2
+        elif a_type == Action.ActionType.PLAY:
+            move_type = 3
+        elif a_type is Action.ActionType.DISCARD:
+            move_type = 4
+        else:
+            move_type = 5  # INVALID_MOVE
+
+        # for hints we fill in `to_` & `card_indices`, for non‐hints it's -1 and empty
+        if move_type in (1, 2):
+            to_ = native_action.pnr
+            card_indices = [native_action.cnr] if native_action.cnr is not None else []
+            card_index = -1
+        else:
+            to_ = -1
+            card_indices = []
+            card_index = native_action.cnr if native_action.cnr is not None else -1
+
+        # `from_` is the active player issuing this action
+        from_ = ...  # TODO: we should get the current player in some way
+
+        # 4) map our Color enum back to the original int code, and rank is just `num`
+        color = COLOR_REVERSE_CONVERSION_DICT[native_action.col]
+        rank = native_action.num
```

Comment:
Since HanaSim's rank is 1-based:
```suggestion
        rank = native_action.num + 1
```